CIP-????/README.md

```diff
@@ -0,0 +1,113 @@
+---
+CIP: ?
+Title: Deterministic universal almost-unique Plutus Constructors
+Category: Plutus
+Status: Proposed
+Authors:
+    - Niels Mündler <niels@opshin.dev>
+Implementors: [Niels Mündler <niels@opshin.dev>]
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/608
+Created: 2023-10-20
+License: CC-BY-4.0
+---
+
+<!-- Existing categories:
+
+- Meta     | For meta-CIPs which typically serves another category or group of categories.
+- Wallets  | For standardisation across wallets (hardware, full-node or light).
+- Tokens   | About tokens (fungible or non-fungible) and minting policies in general.
+- Metadata | For proposals around metadata (on-chain or off-chain).
+- Tools    | A broad category for ecosystem tools not falling into any other category.
+- Plutus   | Changes or additions to Plutus
+- Ledger   | For proposals regarding the Cardano ledger (including Reward Sharing Schemes)
+- Catalyst | For proposals affecting Project Catalyst / the Jörmungandr project
+
+-->
+
+Note: I will use record / Plutus Data exchangibly throughout the document.
+
+## Abstract
+Plutus Constructor IDs are currently heavily focused around their origin in Haskell. They are usually used to distinguish different constructors of a single declared datatype.
+In contrast, one may introduce universally recognized datatypes that are identified by a unique constructor id and can be expected to behave in a specified way (i.e. contain specific fields with specific types).
+For this purpose, we introduce a generic way to compute an almost unique, deterministic and universal constructor id for objects based on their name and field types.
+Note that it is not expected that every language adopts this standard as a default (i.e. for Haskell-like languages there might not be much use of it).
```

Comment:
How is Haskell different here?

Reply:
In Python it is common to declare SoP / Unions after the declaration of the specific types i.e.

```python
class A

class B

class C

AB = Union[A, B]
ABC = Union[A, B, C]
```

In Haskell the definition of the SoP/Union at the same time declares the involved alternatives, hence all involved alternatives are known at the time of declaration

```haskell
data AB = A | B
data ABC = A | B | C -- I guess this would throw an error for redeclaring the constructors A and B?
```